To illustrate a logit binomial regression related to nature, let's create a synthetic dataset where we predict the probability of a certain natural event occurring based on some environmental factors. For simplicity, we'll use a dataset that predicts whether a specific plant species is present in a given area based on various environmental variables.
Scenario
Objective: Predict whether a specific plant species is present (1) or absent (0) based on environmental factors such as temperature, rainfall, and soil pH.
Variables
Presence (Y): Binary outcome variable indicating if the plant species is present (1) or absent (0).
Temperature (X1): Average temperature in degrees Celsius.
Rainfall (X2): Average monthly rainfall in millimeters.
Soil pH (X3): Soil pH level (a continuous variable).

In [11]:
pip install scikit-learn pandas

Note: you may need to restart the kernel to use updated packages.


In [13]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_repo

ImportError: cannot import name 'classification_repo' from 'sklearn.metrics' (C:\Users\User\anaconda3\Lib\site-packages\sklearn\metrics\__init__.py)

In [14]:
# Creating the dataset
data = {
    'Presence': [1, 0, 1, 0, 1, 0, 1, 0, 1, 0],
    'Temperature': [22.5, 18.3, 25.0, 20.0, 23.5, 19.5, 24.0, 21.0, 26.0, 17.5],
    'Rainfall': [120, 80, 150, 90, 130, 70, 140, 85, 160, 75],
    'Soil pH': [6.2, 5.8, 6.5, 5.9, 6.3, 5.7, 6.4, 5.6, 6.6, 5.8]
}

df = pd.DataFrame(data)

# Defining the predictor variables and the response variable
X = df[['Temperature', 'Rainfall', 'Soil pH']]
y = df['Presence']

# Splitting the data into training and test sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Fitting the logistic regression model
model = LogisticRegression()
model.fit(X_train, y_train)

# Making predictions
y_pred = model.predict(X_test)

# Printing the model coefficients
print("Coefficients:", model.coef_)
print("Intercept:", model.intercept_)

# Printing classification report
print("\nClassification Report:\n", classification_report(y_test, y_pred))

Coefficients: [[0.02374921 0.31292028 0.00356309]]
Intercept: [-33.46260445]


NameError: name 'classification_report' is not defined

In [15]:
# Printing the model coefficients
print("Coefficients:", model.coef_)
print("Intercept:", model.intercept_)

# Printing classification report
print("\nClassification Report:\n", classification_report(y_test, y_pred))

Coefficients: [[0.02374921 0.31292028 0.00356309]]
Intercept: [-33.46260445]


NameError: name 'classification_report' is not defined

Coefficients:
Temperature: 0.02374921
Rainfall:  0.31292028
Soil pH: 0.00356309
These coefficients represent the change in the log-odds of the presence of the plant species for each one-unit change in the respective predictor.
Intercept: -33.46260445
This is the log-odds of the plant species being present when all predictors are zero

Scikit-learn: Provides the model coefficients and evaluation metrics, focusing on model performance rather than detailed statistical testing.